lighthouse-core/test/audits/byte-efficiency/bundle-duplication-test.js

```diff
@@ -0,0 +1,224 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const fs = require('fs');
+const BundleDuplicationAudit = require('../../../audits/byte-efficiency/bundle-duplication.js');
+
+function load(name) {
+  const mapJson = fs.readFileSync(
+    `${__dirname}/../../fixtures/source-maps/${name}.js.map`,
+    'utf-8'
+  );
+  const content = fs.readFileSync(`${__dirname}/../../fixtures/source-maps/${name}.js`, 'utf-8');
+  return {map: JSON.parse(mapJson), content};
+}
+
+describe('BundleDuplicationAudit computed artifact', () => {
+  it('works (simple)', async () => {
+    const context = {computedCache: new Map()};
+    const {map, content} = load('foo.min');
+    const artifacts = {
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/foo1.min.js', map},
+        {scriptUrl: 'https://example.com/foo2.min.js', map},
+      ],
+      ScriptElements: [
+        {src: 'https://example.com/foo1.min.js', content},
+        {src: 'https://example.com/foo2.min.js', content},
+      ],
+    };
+    const results = await BundleDuplicationAudit.audit_(artifacts, [], context);
+    expect({items: results.items, wastedBytesByUrl: results.wastedBytesByUrl})
+      .toMatchInlineSnapshot(`
+      Object {
+        "items": Array [
+          Object {
+            "source": "Other",
+            "sourceBytes": Array [],
+            "totalBytes": 0,
+            "url": "",
+            "urls": Array [
+              "https://example.com/foo1.min.js",
+              "https://example.com/foo2.min.js",
+            ],
+            "wastedBytes": 682,
+          },
+        ],
+        "wastedBytesByUrl": Map {
+          "https://example.com/foo2.min.js" => 682,
+        },
+      }
+    `);
+  });
+
+  it('works (complex)', async () => {
+    const context = {computedCache: new Map()};
+    const bundleData1 = load('coursehero-bundle-1');
+    const bundleData2 = load('coursehero-bundle-2');
+    const artifacts = {
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/coursehero-bundle-1.js', map: bundleData1.map},
+        {scriptUrl: 'https://example.com/coursehero-bundle-2.js', map: bundleData2.map},
+      ],
+      ScriptElements: [
+        {src: 'https://example.com/coursehero-bundle-1.js', content: bundleData1.content},
+        {src: 'https://example.com/coursehero-bundle-2.js', content: bundleData2.content},
+      ],
+    };
+    const results = await BundleDuplicationAudit.audit_(artifacts, [], context);
```

Comment:
would love to get 1 test where we exercise the `providedWastedBytesByUrl` too, maybe create some fake bundles for our fixture trace?

Reply:
don't see how to do this ... computeWasteWithTTIGraph just spits out a number. does that method need to be decomposed?

Reply:
I just mean test the results from `audit()` not just `audit_()` at least once